src/App.tsx

```diff
@@ -1,18 +1,42 @@
-import React from 'react';
-import './App.css';
+import { FC, useEffect, useState } from 'react';
+import { Pagination } from './components/Pagination';
+
 import { getNumbers } from './utils';
+import './App.css';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
-export const App: React.FC = () => {
+export const App: FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+
+  const [
+    visibleItems,
+    setVisibleItems,
+  ] = useState(items.slice(0, perPage));
```

Comment:
calculating the initial state value inside the useState() looks ugly a bit 🥲 
I'd create a variable for that